nltk/translate/bleu_score.py

```diff
@@ -479,18 +490,17 @@ def method0(self, p_n, *args, **kwargs):
         p_n_new = []
         _emulate_multibleu = kwargs['emulate_multibleu']
         for i, p_i in enumerate(p_n):
-            if p_i.numerator != 0:
-                p_n_new.append(p_i)
-            elif _emulate_multibleu and i < 5:
-                return [sys.float_info.min]
-            else:
-                _msg = str("\nCorpus/Sentence contains 0 counts of {}-gram overlaps.\n"
-                           "BLEU scores might be undesirable; "
-                           "use SmoothingFunction().").format(i+1)
+            if p_i.numerator == 0:
+                # We have 0/den where den might be 0 or !=0. The result should equal 0, or undefined
+                # However, math.log(0) gives math error. Therefore we replace 0 with sys.float_info.min
+                p_n_new.append(sys.float_info.min)
+                _msg = str("\nThe hypothesis contains 0 counts of {}-gram overlaps. Therefore\n"
+                           "the BLEU score evaluates to 0, independently of how many N-gram\n"
+                           "overlaps of lower order it contains. Consider using BLEU score\n"
```

Comment:
Lowering order of n-grams will require users to get into a wave of other problems that concerns weights hacking. It think it's better to ask users to simply smooth their n-gram precision.